debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -138,63 +136,67 @@ public void execute(ChangeEventSourceContext context) {
-                    LOGGER.debug("\n\n***** SWITCH occurred *****\n" + " from:{} , to:{} \n\n", currentRedoLogFiles, possibleNewCurrentLogFile);
+                    metronome = Metronome.sleeper(Duration.ofMillis(logMinerMetrics.getMillisecondToSleepBetweenMiningQuery()), clock);
+                    metronome.pause();
 
-                    // This is the way to mitigate PGA leak.
-                    // With one mining session it grows and maybe there is another way to flush PGA, but at this point we use new mining session
-                    endMining(connection);
+                    Set<String> possibleNewCurrentLogFile = getCurrentRedoLogFiles(connection, logMinerMetrics);
+                    if (!currentRedoLogFiles.equals(possibleNewCurrentLogFile)) {
+                        LOGGER.debug("\n\n***** SWITCH occurred *****\n" + " from:{} , to:{} \n\n", currentRedoLogFiles, possibleNewCurrentLogFile);
 
-                    if (!isContinuousMining) {
-                        if (strategy == OracleConnectorConfig.LogMiningStrategy.CATALOG_IN_REDO) {
-                            buildDataDictionary(connection);
+                        // This is the way to mitigate PGA leak.
+                        // With one mining session it grows and maybe there is another way to flush PGA, but at this point we use new mining session
+                        endMining(connection);
+
+                        if (!isContinuousMining) {
+                            if (strategy == OracleConnectorConfig.LogMiningStrategy.CATALOG_IN_REDO) {
+                                buildDataDictionary(connection);
+                            }
+
+                            abandonOldTransactionsIfExist(connection);
+                            setRedoLogFilesForMining(connection, startScn);
                         }
 
-                        abandonOldTransactionsIfExist(connection);
-                        setRedoLogFilesForMining(connection, startScn);
+                        currentRedoLogFiles = getCurrentRedoLogFiles(connection, logMinerMetrics);
                     }
 
-                    currentRedoLogFiles = getCurrentRedoLogFiles(connection, logMinerMetrics);
-                }
+                    startOnlineMining(connection, startScn, endScn, strategy, isContinuousMining);
 
-                startOnlineMining(connection, startScn, endScn, strategy, isContinuousMining);
+                    Instant startTime = Instant.now();
+                    fetchFromMiningView.setFetchSize(10_000);
+                    fetchFromMiningView.setLong(1, startScn);
+                    fetchFromMiningView.setLong(2, endScn);
 
-                Instant startTime = Instant.now();
-                fetchFromMiningView.setFetchSize(10_000);
-                fetchFromMiningView.setLong(1, startScn);
-                fetchFromMiningView.setLong(2, endScn);
+                    ResultSet res = fetchFromMiningView.executeQuery();
```

Comment:
Is it possible to add the result set into try with resources?

Reply:
Fixed